devops/continuous_integration/tutorial.rst

```diff
@@ -0,0 +1,141 @@
+is it going to break the existing ``game/1.0`` and/or ``mapviewer/1.0`` applications? Is it necessary to re-build from source some of the existing
+packages that depend directly or indirectly on ``ai`` package? In this tutorial we will use ``game/1.0`` and ``mapviewer/1.0`` as our "products",
+but this concept will be further explained later, and specially why it is important to think in terms of "products" instead of trying to explicitly
+model the dependencies top-bottom in the CI.
+
+
+Repositories and promotions
+---------------------------
+
+The concept of multiple server side repositories is very important for CI. In this tutorial we will use 3 repositories:
+
+- ``develop``: This repository is the main one that developers have configured in their machines to be able to ``conan install`` dependencies
+  and work. As such it is expected to be quite stable, similar to a shared "develop" branch in git, and the repository should contain pre-compiled
+  binaries for the organization pre-defined platforms, so developers and CI don't need to do ``--build=missing`` and build again and again from
+  source.
+- ``packages``: This repository will be used to upload individual package binaries for different configurations. To consider a certain change
+  in a package source code to be correct, it might require that such change build correctly under a variaty of platforms, lets say Windows and Linux.
+  If the package builds correctly under Linux more quickly, we can upload it to the ``packages`` repository, and wait until the Windows build 
+  finishes, and only when both are correct we can proceed. The ``packages`` repository serves as a temporary storage when building different
+  binaries for the same package in different platforms concurrently, until all of those configurations have been built correctly. In the example
+  above, when some developer did source changes in a new ``ai/1.1.0`` recipe, the different binaries for Windows and Linux will be built in different
+  servers. These jobs can upload their respective binaries for Windows and Linux to the ``packages`` binary repository. Note that these individual
+  binaries will not disrupt other developers or CI jobs, as they don't use the ``packages`` repository.
+- ``products``: It is possible that some changes create new package versions or revisions correctly. But these new versions might break consumers
+  of those packages, for example some changes in the new ``ai/1.1.0`` package might unexpectedly break ``engine/1.0``. Or even if they don't
+  necessarily break, they might still need to build a new binary from source for ``engine/1.0`` and/or ``game/1.0``. The ``products`` binary
+  repository will be the place where binaries for different packages are uploaded to not disrupt or break the ``develop`` repository, until
+  the "products pipeline" can build necessary binaries from source and verify that these packages integrate cleanly.
```

Comment:
I have the impression that this would be better if summarized in bullet points or something like that, maybe just giving the most important attributes of each server and later readers will have more data when they read the full story? Or at least summarize a bit the ``packages`` that I think ends up being hard to follow.

Reply:
I think that all this part maybe is hard to follow because is trying to advance too much about the story, maybe we should limit here to explain briefly what kind of binaries is storing each repo and then making sense of it later?

Reply:
Done